natss/pkg/dispatcher/dispatcher.go

```diff
@@ -69,16 +72,22 @@ type SubscriptionsSupervisor struct {
 	hostToChannelMap atomic.Value
 }
 
-// NewDispatcher returns a new SubscriptionsSupervisor.
-func NewDispatcher(natssURL, clusterID, clientID string, logger *zap.Logger) (*SubscriptionsSupervisor, error) {
+type NatssDispatcher interface {
+	Start(stopCh <-chan struct{}) error
+	UpdateSubscriptions(channel *messagingv1alpha1.Channel, isFinalizer bool) (map[eventingduck.SubscriberSpec]error, error)
+	UpdateHostToChannelMap(ctx context.Context, chanList []messagingv1alpha1.Channel) error
```

Comment:
As discussed in person: You are right. However, it is just an interface wrapped over the dispatcher in order to create a mock. Therefore I took all public methods from the dispatcher and made an interface out of it. I agree that the name is highly implementation specific. The best would probably be to move the listing of the natsschannel to the dispatcher. Then we could remove the method from the interface at all. But I don't wanna change the implementation of this in this PR.

For now, what about renaming it to `ProcessChannels` ? Because it is a method which takes all natsschannel and does some internal bookkeeping.

Reply:
I like that name better, it's more generic.